judge/judgedaemon.main.php

```diff
@@ -418,13 +418,15 @@ function fetch_executable_internal(
                         // no main class detection here
                         $buildscript .= "echo 'COMPARE_DIR=\$(dirname \"\$0\")' >> run\n";
                         $mainClass = basename($unescapedSource, '.java');
-                        $buildscript .= "echo 'java -cp \"\$COMPARE_DIR\" $mainClass \"\\\$@\"' >> run\n";
+                        // Note: since the $@ is within single quotes, we do not need to double escape it.
+                        $buildscript .= "echo 'java -cp \"\$COMPARE_DIR\" $mainClass \"\$@\"' >> run\n";
                         $buildscript .= "chmod +x run\n";
                         break;
                     case 'py':
                         $buildscript .= "echo '#!/bin/sh' > run\n";
                         $buildscript .= "echo 'COMPARE_DIR=\$(dirname \"\$0\")' >> run\n";
-                        $buildscript .= "echo 'python3 \$COMPARE_DIR/$source' \"\\\$@\" >> run\n";
+                        // Note: since the $@ is within single quotes, we do not need to double escape it.
+                        $buildscript .= "echo 'python3 \$COMPARE_DIR/$source' \"\$@\" >> run\n";
```

Comment:
If I am not missing anything, this is still broken and actually python verifier might have been the reason why you did https://github.com/DOMjudge/domjudge/commit/3e65d9805a6e14eb0ad947ede7ee57d0ef7f7984 

Back then you probably realized that python is broken and then "fixed" Java as well. Now, I assume PC2 had Java validators and you "fixed" python as well?

So, if you look closely, then for Java the "whole" `echo` args are surrounded by `'` while for python the final `\"\$@\"` is not quoted.

Aligning both to the Java style is the best course of action IMHO.

Reply:
Ah you are right. NAC had Python scripts and that broke. Let me make them in sync.

Reply:
Done!